garble.py

```diff
@@ -70,7 +73,22 @@ def validate_clks(clk_files, metadata_file):
 
 def garble_pii(args):
     secret_file = Path(args.secretfile)
-    source_file = Path(args.sourcefile)
+
+    if args.sourcefile:
+        source_file = Path(args.sourcefile)
+    else:
+        oldest_ts = datetime.fromtimestamp(0)
+        oldest_name = ""
+        for filename in filter(
+            lambda x: "pii" in x and len(x) == 23, os.listdir("temp-data")
+        ):
+            timestamp = datetime.strptime(filename[4:-4], "%Y%m%dT%H%M%S")
```

Comment:
I'm not sure it's worth changing but the YYYYMMDD format means you could just take the maximum. The benefit of parsing I suppose is that you crash if there's anything unexpected in there.

Reply:
are you suggesting I do `max()` on the timestamps as strings? I see that how/why this would work (and would even work with the current format because the `%H%M%S` format gives 24-hr time but would prefer at least parsing the time to make sure it's a time and we're not accidentally comparing an errant file that ended up in `temp-data`. A hypothetical `"pii-data-report-v2.json"` which would also be picked up by this code and would str-compare to be newer than all of the timestamps because of how python compares strings. That said I could replace a few lines of this logic with a `max()` of the parsed timestamp objects, because `datetime` has comparators built in to it. Does that make sense?

Reply:
If you'd prefer something that's more crash-resistant, I could filter on a regex to ensure the filename is exactly what we're looking for but to me that felt like a little bit overkill, but I'm happy to put it in if you think it's warranted.